docs/src/experimental-logically-bound-images.md

```diff
@@ -28,8 +32,17 @@ RUN ln -s /usr/share/containers/systemd/my-app.image /usr/lib/bootc/bound-images
     ln -s /usr/share/containers/systemd/my-app.image /usr/lib/bootc/bound-images.d/my-app.image
 ```
 
+In the `.container` definition, you should use:
+
+```
+GlobalArgs=--storage-opt=additionalimagestore=/usr/lib/bootc/storage
```

Comment:
Couldn't we globally configure the additional image store via containers-storage.conf?  That would quite improve the UX.

Reply:
Yes, I think it's a possibility. That said, changing the storage configuration without having https://github.com/containers/storage/pull/1885 is...a bit ugly.

There are a few other things that have me slightly wary:

- All of this code is basically a no-op unless bound images are in use, but if we added an additional store it would mean every `podman` invocation would see it...it'd be empty, but it would be a global behavioral change
- There's been some lingering concerns I've seen in the past around what happens if...

Yeah ok, I think this a near showstopper for *globally* (i.e. in `storage.conf`) setting additionalimagestores (hereafter: AIS) as currently defined today. Basically, if an AIS has an image/layer, when pulling it will be reused. Makes sense!

But today, the bootc bound images logic will prune images that are no longer referenced.

This should work fine for bootc logically bound images (hereafter: LBI) because *all* layers in the stack live in the bootc store; podman sees every layer as read-only.

But yes, I just tested this, and things explode (as you'd expect) in the scenario where a lower read-only layer disappears for the case of an image/container that lives in `/var/lib/containers`.

Reply:
I guess of course, bootc's GC logic could learn to take into account the `/var/lib/containers` store state...but we're starting to "cross the worlds" with stuff like that.

Reply:
Now, one might be thinking...isn't the real fix for this to have *one storage* by default, i.e. on bootc systems podman and bootc default to having the same storage for images (underneath `/sysroot`, though containers would probably still make sense under `/var).

And yes maybe, but the thing that terrified me about going down that path is that typing "podman ..." has the potential to break the operating system. There's lots of docs talking about `podman system reset` and the like.

Could we make it reliable? Yes, clearly. But it'd require I think a lot of careful thought, code auditing, cooperation and integration.

Reply:
It would be nice if podman had a way to recover from it. IE Pull down the missing image but I am not sure the missing image/layer would still exist or have a way to discover where it exist.

Reply:
we had agreed to create a pinning image mechanism for you that was going to avoid the reset problem no?  i thought that is the way we were going?

Reply:
I am not aware of any commitment from anyone to work on image pinning on the podman side.

That said again, I should be very clear that my thoughts on this design changed as we got into the implementation. My original thought was basically "bootc is just running podman pull, how hard can it be Michael? 1 week?"

But yeah...having bootc touch `/var` (`/var/lib/containers`) just blows up our story around state, messes with transactional updates, partitioning, etc. So that led me rapidly down this path of having a bootc-owned storage.

Reply:
Also don't we want the ability to roll back versions of images?

But bottom line we have the same issue if anyone sets up additional stores. If an image is used from an additional store and gets removed, then the container will no longer work. In RHIVOS we thought about this and came to the conclusion that we would know which images were being used so we would know what images to remove.  Also containers are ephemeral, they did not survive reboot.

Reply:
> Also don't we want the ability to roll back versions of images?

bootc retains all LBIs that are referenced from any deployment, i.e. when you upgrade but then roll back, you are guaranteed that the LBIs are still there.

(However, there's ill-defined semantics currently around how LBIs work with floating tags, but basically "don't do that")